packages/core/src/application.ts

```diff
@@ -161,6 +161,36 @@ export class Application extends Context {
     );
   }
 
+  // tslint:disable-next-line:no-any
+  public config(cfg: {[key: string]: any}) {
+    for (const key in cfg) {
+      const item = cfg[key];
+      // Iterate through the keys one level down (not recursively!)
+      if (item instanceof Object) {
+        for (const subkey in item) {
+          this.bindItemOrClass(`${key}.${subkey}`, item[subkey]);
+        }
+      }
+      this.bindItemOrClass(key, item);
```

Comment:
Do we want to add a namespace to avoid possible conflicts with other binding providers? For example, `app.config.<key>`?

Reply:
No, in fact, one of the awesome parts about this is that it can be used as your one-stop shop for wiring up _everything_ if you want; the unit test actually demonstrates using this to wire up a named server instance.

Reply:
(Which will also require no code changes at this end when we refactor for multiple server support!)

Reply:
The test should probably change when we do that, though, but it'll end up being a part of that PR, if this lands first.